### DISCUSSION TITLE
Find option in Binary editor?

### DISCUSSION THREAD

**Discussion by pskowronek** [mucommander/mucommander#914] [Ideas]
A discussion started here: https://github.com/mucommander/mucommander/pull/908#issuecomment-1519184163

**Comment by ahadas** [comment]
@hajdam ^

**Comment by hajdam** [comment]
I'll try to add that - now that pull request for find in text editor is merged... I'll go with simple dialog variant thou. No search bar or regular expressions :-|

**Reply by pskowronek** [reply]
> I'll try to add that - now that pull request for find in text editor is merged... I'll go with simple dialog variant thou. No search bar or regular expressions :-|

You may want to update this file: https://github.com/mucommander/mucommander/blob/d02d2784e0b069afe87b54a1ee122425d55abd35/mucommander-viewer-binary/src/main/java/com/mucommander/viewer/binary/BinaryFileViewerService.java#L68 so you could automatically call Find when BinaryEditor is opened (use `SearchProperty` enum to get Search File with content details). See https://github.com/mucommander/mucommander/pull/906/.

**Reply by hajdam** [reply]
> You may want to update this file:
> 
> https://github.com/mucommander/mucommander/blob/d02d2784e0b069afe87b54a1ee122425d55abd35/mucommander-viewer-binary/src/main/java/com/mucommander/viewer/binary/BinaryFileViewerService.java#L68
> so you could automatically call Find when BinaryEditor is opened (use `SearchProperty` enum to get Search File with content details). See #906.

Ugh... Very ugly boolean argument in my opinion :-( But as you wish...

**Reply by pskowronek** [reply]
> > You may want to update this file:
> > https://github.com/mucommander/mucommander/blob/d02d2784e0b069afe87b54a1ee122425d55abd35/mucommander-viewer-binary/src/main/java/com/mucommander/viewer/binary/BinaryFileViewerService.java#L68
> > 
> > so you could automatically call Find when BinaryEditor is opened (use `SearchProperty` enum to get Search File with content details). See #906.
> 
> Ugh... Very ugly boolean argument in my opinion :-( But as you wish...

yup, I know. You may want to refactor or suggest how you would see it - I may do it. Enum?

**Reply by hajdam** [reply]
You don't have to - I would question too much about it :-P It's not just that it's boolean, but also name "fromSearchWithContent" - it's like class should be aware of some external circumstances it has not control over at all and do some magic, instead of describing what the parameter do. Not sure if it should be even in constructor and actual related value is red from SearchProperty singleton which I don't like as well... I'll just adapt - my code is mostly mess anyway...
